dvc/repo/fetch.py

```diff
@@ -22,7 +22,11 @@ def fetch(
             remote=remote,
             jobs=jobs,
             recursive=recursive,
-        )["local"]
-        return self.cloud.pull(
-            used, jobs, remote=remote, show_checksums=show_checksums
         )
+
+        for stage in used["repo"]:
+            stage.reproduce()
```

Comment:
Need to `stage.unlock()` btw. But really, you don't even need to do that, because it seems like you could simply collect dependencies in used['repo'] and then `dep.fetch()` here.